docs: password policy maximum character enforcement PEM-8603 (#8334)
spectrocloud/librarium: docs/docs-content/tenant-settings/password-policy.md

---
sidebar_label: "Password Policy"
title: "Password Policy"
description: "Learn how to configure the password policy for your Palette tenant."
icon: ""
hide_table_of_contents: false
sidebar_position: 50
tags: ["tenant-administration", "security", "password"]
---

Tenant administrators can update the password policy for Palette users. The default password policy is as follows.

> - The password must be at least 6 characters long.

Reviewer: ```suggestion
- The password must be at least six characters long.
```
Our Style guide [suggests](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1765933057/Spectro+Cloud+Internal+Style+Guide#Numbers) we spell out numbers 0-9. 
I didn't suggest spelling numbers in the table below though, cause the users will be entering numbers.

> - The password must be at most 64 characters long.

Reviewer 2: Two thoughts:
1. "At most" feels vague. Shouldn't it be "max"?
2. line 45 says "max 128 charaters". So is it 64 or 128 characters?

Author: I went with at most because it complements at least.

- The password expires after 365 days.
- The password must contain at least one uppercase letter.
- The password must contain at least one lowercase letter.
- The password must contain at least one digit.
- The password must contain at least one special character.

> If you want to change the default password, follow the steps below.

Reviewer: ```suggestion
If you want to change the default password policy, follow the steps below.
```

## Prerequisites

- Palette tenant admin access.

## Update Password Policy

1. Log in to [Palette](https://console.spectrocloud.com) as a tenant admin.

> 2. Navigate to the left **Main Menu** and select **Tenant Settings**.

Reviewer: ```suggestion
2. Navigate to the left main menu and select **Tenant Settings**.
```

According to the [Style guide](https://spectrocloud.atlassian.net/wiki/spaces/DE/pages/1765933057/Spectro+Cloud+Internal+Style+Guide):

> UI elements that do not have text associated with them, such as left main menu, three-dot menu, and drop-down menu, are neither bolded nor treated as proper nouns.

3. From the **Tenant Settings Menu**, select **Password Policy**.

> 4. Configure the password policy using the configuration fields.

Reviewer: ```suggestion
4. Set up the password policy using the configuration fields.
```
Maybe? To avoid repeating "Configure... configuration"

   | **Configuration**                    | **Description**                                                                                                                             |
   | ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------- |
   | **Enable Regex**                     | Enable if you want to specify the password policy using a regular expression.                                                               |
   | **Regex**                            | Specify a regular expression that the passwords must conform to.                                                                            |

>    | **Expiry Duration**                  | Set a password expiry time. Select from **3 months**, **6 months** and **Custom**.                                                          |

Reviewer: ```suggestion
   | **Expiry Duration**                  | Set a password expiry time. Select from **3 months**, **6 months**, and **Custom**.                                                          |
```
Oxford comma

>    | **Expiry Duration (Days)**           | If you have selected custom expiry duration, specify a number of days for password expiry.                                                  |

Reviewer: ```suggestion
   | **Expiry Duration (Days)**           | If you have selected **Custom** expiry duration, specify a number of days for password expiry.                                                  |
```
To highlight there is a corresponding text in the UI

>    | **First Reminder (Days)**            | Specify when to send out a password expiry reminder email to users.                                                                         |

Reviewer: ```suggestion
   | **First Reminder (Days)**            | Specify the number of days before the password expires to send the first reminder email to users.|
```

How about this? I think it's a bit more precise.

   | **Min Length**                       | Specify the minimum length of the password. The configured value must be between 6 and 128 and should not exceed the **Max Length**.        |

>    | **Max Length**                       | Specify the maximum length of the password. The configured value must be between 6 and 128 and should not be less than then **Min Length**. |

Reviewer: ```suggestion
   | **Max Length**                       | Specify the maximum length of the password. The configured value must be between 6 and 128 and should not be less than the **Min Length**. |
```

   | **Min Number Of Block Letters**      | Specify the minimum number of uppercase letters in the password. The configured value must be larger than 1.                                |
   | **Min Number Of Digits**             | Specify the minimum number of digits in the password. The configured value must be larger than 1.                                           |
   | **Min Number Of Small Letters**      | Specify the minimum number of lowercase letters in the password. The configured value must be larger than 1.                                |
   | **Min Number Of Special Characters** | Specify the minimum number of special characters in the password. The configured value must be larger than 1.                               |

> 5. Click on **Save** to save the changes.

Reviewer: ```suggestion
5. Click on **Save** to apply the changes.
```
